In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import pyarrow.parquet as pq
from datetime import datetime as dt

1. import all the necessary packages from the python library
2. import the pyarrow.parquet from the python library since the data set of the train file is in parquet format.

In [2]:
!pip install pandas pyarrow


[notice] A new release of pip available: 22.2.1 -> 23.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip


3. Install the pandas pyarrow library to import the parquet file into the pandas DataFrame

In [3]:
df = pd.read_parquet("C:\\Users\\thiru\\OneDrive\\Desktop\\Akaike Technologies\\Structured_Data_Assignment\\Structured_Data_Assignment\\train.parquet")
df.head()

Patient-Uid       Date           Incident
0  a0db1e73-1c7c-11ec-ae39-16262ee38c7f 2019-03-09  PRIMARY_DIAGNOSIS
1  a0dc93f2-1c7c-11ec-9cd2-16262ee38c7f 2015-05-16  PRIMARY_DIAGNOSIS
3  a0dc94c6-1c7c-11ec-a3a0-16262ee38c7f 2018-01-30     SYMPTOM_TYPE_0
4  a0dc950b-1c7c-11ec-b6ec-16262ee38c7f 2015-04-22        DRUG_TYPE_0
8  a0dc9543-1c7c-11ec-bb63-16262ee38c7f 2016-06-18        DRUG_TYPE_1

4. use pd.read_parquet to get the dataset as pandas DataFrame.

In [4]:
df.to_csv("train.csv")

In [5]:
df.tail()

Patient-Uid       Date      Incident
29080886  a0ee9f75-1c7c-11ec-94c7-16262ee38c7f 2018-07-06   DRUG_TYPE_6
29080897  a0ee1284-1c7c-11ec-a3d5-16262ee38c7f 2017-12-29   DRUG_TYPE_6
29080900  a0ee9b26-1c7c-11ec-8a40-16262ee38c7f 2018-10-18  DRUG_TYPE_10
29080903  a0ee1a92-1c7c-11ec-8341-16262ee38c7f 2015-09-18   DRUG_TYPE_6
29080911  a0ee146e-1c7c-11ec-baee-16262ee38c7f 2018-10-05   DRUG_TYPE_1

In [6]:
df.isnull().sum()

Patient-Uid    0
Date           0
Incident       0
dtype: int64

There is no null value in the train dataset

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3220868 entries, 0 to 29080911
Data columns (total 3 columns):
 #   Column       Dtype         
---  ------       -----         
 0   Patient-Uid  object        
 1   Date         datetime64[ns]
 2   Incident     object        
dtypes: datetime64[ns](1), object(2)
memory usage: 98.3+ MB


5. Patient-Uid feature data type is object
6. Date feature is in datetime64[ns]
7. Incident feature is data type is object

In [10]:
df.value_counts()

Patient-Uid                           Date        Incident   
a0efafa8-1c7c-11ec-9d1b-16262ee38c7f  2020-01-07  DRUG_TYPE_6    7
a0df8e7b-1c7c-11ec-b86c-16262ee38c7f  2017-02-03  DRUG_TYPE_6    7
a0e0d623-1c7c-11ec-bac3-16262ee38c7f  2020-05-14  DRUG_TYPE_6    7
a0efc2b9-1c7c-11ec-acbe-16262ee38c7f  2020-03-20  DRUG_TYPE_6    6
a0ef9541-1c7c-11ec-b3fb-16262ee38c7f  2019-05-01  DRUG_TYPE_6    6
                                                                ..
a0e3e997-1c7c-11ec-8f39-16262ee38c7f  2018-11-28  DRUG_TYPE_7    1
                                      2018-12-15  TEST_TYPE_1    1
                                      2019-02-14  DRUG_TYPE_1    1
                                      2019-03-13  DRUG_TYPE_0    1
a0f0d582-1c7c-11ec-a6c1-16262ee38c7f  2020-08-08  DRUG_TYPE_5    1
Length: 3185297, dtype: int64

In [11]:
shape = df.shape
print("Total number of rows:",shape[0]);
print("Total number of columns:",shape[1]);


Total number of rows: 3220868
Total number of columns: 3


In [12]:
df.describe().transpose()

C:\Users\thiru\AppData\Local\Temp\ipykernel_20972\609525639.py:1: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
  df.describe().transpose()


count unique                                   top    freq  \
Patient-Uid  3220868  27033  a0ddfd2c-1c7c-11ec-876d-16262ee38c7f    1645   
Date         3220868   1977                   2019-05-21 00:00:00    3678   
Incident     3220868     57                           DRUG_TYPE_6  561934   

                 first       last  
Patient-Uid        NaT        NaT  
Date        2015-04-07 2020-09-03  
Incident           NaT        NaT

from above table we can find there are total 27,033 unique paitent-ids

In [14]:
df["Incident"].value_counts()

DRUG_TYPE_6          561934
DRUG_TYPE_1          484666
PRIMARY_DIAGNOSIS    431902
DRUG_TYPE_0          300005
DRUG_TYPE_7          258782
DRUG_TYPE_2          256841
DRUG_TYPE_8          160066
DRUG_TYPE_3          127676
TEST_TYPE_1           96810
TARGET DRUG           67218
DRUG_TYPE_9           66894
DRUG_TYPE_5           57510
DRUG_TYPE_11          48118
SYMPTOM_TYPE_0        46078
SYMPTOM_TYPE_6        32066
TEST_TYPE_0           27570
SYMPTOM_TYPE_7        22019
DRUG_TYPE_10          20925
DRUG_TYPE_14          17306
DRUG_TYPE_13          12372
DRUG_TYPE_12           9551
SYMPTOM_TYPE_14        8927
SYMPTOM_TYPE_1         8608
SYMPTOM_TYPE_2         8168
TEST_TYPE_3            8115
SYMPTOM_TYPE_5         7583
SYMPTOM_TYPE_8         7430
TEST_TYPE_2            7021
SYMPTOM_TYPE_15        6295
SYMPTOM_TYPE_10        6005
SYMPTOM_TYPE_29        5950
SYMPTOM_TYPE_16        4940
DRUG_TYPE_15           4906
SYMPTOM_TYPE_9         4885
DRUG_TYPE_4            4566
SYMPTOM_TYPE_4      

Group based on the target drug and get the unique patient id and their last target drug date


In [54]:
df["Incident"].nunique()

57

Total 57 unique incident found in the record

In [16]:
df.groupby(["Patient-Uid","Date"])["Incident"].count

<bound method GroupBy.count of <pandas.core.groupby.generic.SeriesGroupBy object at 0x000001F601F9F610>>

In [17]:
dp=df.loc[df["Incident"]=="TARGET DRUG"]
dp.head()

Patient-Uid       Date     Incident
3294791  a0eb742b-1c7c-11ec-8f61-16262ee38c7f 2020-04-09  TARGET DRUG
3296990  a0edaf09-1c7c-11ec-a360-16262ee38c7f 2018-06-12  TARGET DRUG
3305387  a0e9fa0e-1c7c-11ec-8dc7-16262ee38c7f 2019-06-11  TARGET DRUG
3309423  a0ecc615-1c7c-11ec-aa31-16262ee38c7f 2019-11-15  TARGET DRUG
3309494  a0ea612f-1c7c-11ec-8cf0-16262ee38c7f 2020-03-18  TARGET DRUG

Grouping the data set base on the incident "TARGET DRUG"

In [18]:
dp.groupby(["Patient-Uid"])["Date"].max()

Patient-Uid
a0e9c384-1c7c-11ec-81a0-16262ee38c7f   2020-09-02
a0e9c3b3-1c7c-11ec-ae8e-16262ee38c7f   2019-10-29
a0e9c3e3-1c7c-11ec-a8b9-16262ee38c7f   2020-05-26
a0e9c414-1c7c-11ec-889a-16262ee38c7f   2019-11-22
a0e9c443-1c7c-11ec-9eb0-16262ee38c7f   2020-07-14
                                          ...    
a0f0d4c5-1c7c-11ec-bfec-16262ee38c7f   2020-08-05
a0f0d4f4-1c7c-11ec-b144-16262ee38c7f   2020-07-18
a0f0d523-1c7c-11ec-89d2-16262ee38c7f   2020-08-11
a0f0d553-1c7c-11ec-a70a-16262ee38c7f   2020-07-21
a0f0d582-1c7c-11ec-a6c1-16262ee38c7f   2020-08-05
Name: Date, Length: 9374, dtype: datetime64[ns]

In [19]:
dp.shape

(67218, 3)

In [20]:
print("Total person who has taken target drugs:",dp.shape[0])


Total person who has taken target drugs: 67218


In [22]:
dp.head()

Patient-Uid       Date     Incident
3294791  a0eb742b-1c7c-11ec-8f61-16262ee38c7f 2020-04-09  TARGET DRUG
3296990  a0edaf09-1c7c-11ec-a360-16262ee38c7f 2018-06-12  TARGET DRUG
3305387  a0e9fa0e-1c7c-11ec-8dc7-16262ee38c7f 2019-06-11  TARGET DRUG
3309423  a0ecc615-1c7c-11ec-aa31-16262ee38c7f 2019-11-15  TARGET DRUG
3309494  a0ea612f-1c7c-11ec-8cf0-16262ee38c7f 2020-03-18  TARGET DRUG

In [23]:
dp_unique=dp.drop_duplicates(subset=["Patient-Uid"])
dp_unique.head()

Patient-Uid       Date     Incident
3294791  a0eb742b-1c7c-11ec-8f61-16262ee38c7f 2020-04-09  TARGET DRUG
3296990  a0edaf09-1c7c-11ec-a360-16262ee38c7f 2018-06-12  TARGET DRUG
3305387  a0e9fa0e-1c7c-11ec-8dc7-16262ee38c7f 2019-06-11  TARGET DRUG
3309423  a0ecc615-1c7c-11ec-aa31-16262ee38c7f 2019-11-15  TARGET DRUG
3309494  a0ea612f-1c7c-11ec-8cf0-16262ee38c7f 2020-03-18  TARGET DRUG

Dropping the duplicate to get the only the unique Paitent-Uid

In [24]:
print("Total unique Patient-Uid :",dp_unique.shape[0])

Total unique Patient-Uid : 9374


In [25]:
intial_dose_date = dp_unique.groupby("Patient-Uid")["Date"].min().reset_index()
intial_dose_date.head()

Patient-Uid       Date
0  a0e9c384-1c7c-11ec-81a0-16262ee38c7f 2020-08-05
1  a0e9c3b3-1c7c-11ec-ae8e-16262ee38c7f 2018-08-09
2  a0e9c3e3-1c7c-11ec-a8b9-16262ee38c7f 2020-05-26
3  a0e9c414-1c7c-11ec-889a-16262ee38c7f 2019-02-22
4  a0e9c443-1c7c-11ec-9eb0-16262ee38c7f 2019-12-06

 To find the intial "Traget drug" dose days, We are grouping the paitent-uid and the first date of the target drug used for a paitent.

In [26]:
dn= df.merge(intial_dose_date,on="Patient-Uid",how="left")
dn.head()

Patient-Uid     Date_x           Incident Date_y
0  a0db1e73-1c7c-11ec-ae39-16262ee38c7f 2019-03-09  PRIMARY_DIAGNOSIS    NaT
1  a0dc93f2-1c7c-11ec-9cd2-16262ee38c7f 2015-05-16  PRIMARY_DIAGNOSIS    NaT
2  a0dc94c6-1c7c-11ec-a3a0-16262ee38c7f 2018-01-30     SYMPTOM_TYPE_0    NaT
3  a0dc950b-1c7c-11ec-b6ec-16262ee38c7f 2015-04-22        DRUG_TYPE_0    NaT
4  a0dc9543-1c7c-11ec-bb63-16262ee38c7f 2016-06-18        DRUG_TYPE_1    NaT

In [27]:
dn["Date_y"].value_counts()

2020-08-19    6393
2020-06-02    6178
2020-05-19    5437
2020-07-21    5309
2020-08-12    5305
              ... 
2019-05-13      59
2020-07-20      58
2020-06-21      57
2020-02-28      57
2020-02-03      52
Name: Date_y, Length: 786, dtype: int64

In [28]:
dn=dn[dn["Date_y"].isnull()].sample(n=len(dp_unique),random_state=42)
dn.head()

Patient-Uid     Date_x         Incident  \
157156   a0df9e0a-1c7c-11ec-852c-16262ee38c7f 2017-12-14      DRUG_TYPE_1   
1747636  a0de978d-1c7c-11ec-9221-16262ee38c7f 2018-03-06      DRUG_TYPE_1   
602750   a0de9122-1c7c-11ec-b0d7-16262ee38c7f 2020-08-10  SYMPTOM_TYPE_29   
1310089  a0e6fc48-1c7c-11ec-ae9d-16262ee38c7f 2019-02-12      DRUG_TYPE_1   
1183222  a0de5e68-1c7c-11ec-ad5a-16262ee38c7f 2020-07-17      DRUG_TYPE_6   

        Date_y  
157156     NaT  
1747636    NaT  
602750     NaT  
1310089    NaT  
1183222    NaT

In [29]:
con_data = pd.concat([dp_unique,dn])
con_data.head()

Patient-Uid       Date     Incident Date_x  \
3294791  a0eb742b-1c7c-11ec-8f61-16262ee38c7f 2020-04-09  TARGET DRUG    NaT   
3296990  a0edaf09-1c7c-11ec-a360-16262ee38c7f 2018-06-12  TARGET DRUG    NaT   
3305387  a0e9fa0e-1c7c-11ec-8dc7-16262ee38c7f 2019-06-11  TARGET DRUG    NaT   
3309423  a0ecc615-1c7c-11ec-aa31-16262ee38c7f 2019-11-15  TARGET DRUG    NaT   
3309494  a0ea612f-1c7c-11ec-8cf0-16262ee38c7f 2020-03-18  TARGET DRUG    NaT   

        Date_y  
3294791    NaT  
3296990    NaT  
3305387    NaT  
3309423    NaT  
3309494    NaT

In [30]:
con_data["Date_x"].value_counts()

2019-05-22    18
2020-01-22    15
2019-03-16    14
2016-01-21    14
2016-07-14    13
              ..
2015-08-04     1
2019-12-29     1
2019-01-19     1
2019-03-25     1
2015-11-30     1
Name: Date_x, Length: 1895, dtype: int64

In [31]:
con_data["Date_y"].value_counts()

Series([], Name: Date_y, dtype: int64)

In [32]:
frequency_based= con_data.groupby("Patient-Uid")["Incident"].value_counts().unstack().fillna(0);
frequency_based.head()


Incident                              DRUG_TYPE_0  DRUG_TYPE_1  DRUG_TYPE_10  \
Patient-Uid                                                                    
a0dc93f2-1c7c-11ec-9cd2-16262ee38c7f          0.0          0.0           0.0   
a0dc94c6-1c7c-11ec-a3a0-16262ee38c7f          0.0          0.0           0.0   
a0dc9543-1c7c-11ec-bb63-16262ee38c7f          0.0          0.0           0.0   
a0dc95f9-1c7c-11ec-b968-16262ee38c7f          0.0          0.0           0.0   
a0dc9659-1c7c-11ec-a91e-16262ee38c7f          0.0          2.0           0.0   

Incident                              DRUG_TYPE_11  DRUG_TYPE_12  \
Patient-Uid                                                        
a0dc93f2-1c7c-11ec-9cd2-16262ee38c7f           0.0           0.0   
a0dc94c6-1c7c-11ec-a3a0-16262ee38c7f           0.0           0.0   
a0dc9543-1c7c-11ec-bb63-16262ee38c7f           1.0           0.0   
a0dc95f9-1c7c-11ec-b968-16262ee38c7f           0.0           0.0   
a0dc9659-1c7c-11ec-a91e-16262ee38c7f           0.0           0.0   

Incident                              DRUG_TYPE_13  DRUG_TYPE_14  \
Patient-Uid                                                        
a0dc93f2-1c7c-11ec-9cd2-16262ee38c7f           0.0           0.0   
a0dc94c6-1c7c-11ec-a3a0-16262ee38c7f           0.0           0.0   
a0dc9543-1c7c-11ec-bb63-16262ee38c7f           0.0           0.0   
a0dc95f9-1c7c-11ec-b968-16262ee38c7f           0.0           0.0   
a0dc9659-1c7c-11ec-a91e-16262ee38c7f           0.0           0.0   

Incident                              DRUG_TYPE_15  DRUG_TYPE_16  \
Patient-Uid                                                        
a0dc93f2-1c7c-11ec-9cd2-16262ee38c7f           0.0           0.0   
a0dc94c6-1c7c-11ec-a3a0-16262ee38c7f           0.0           0.0   
a0dc9543-1c7c-11ec-bb63-16262ee38c7f           0.0           0.0   
a0dc95f9-1c7c-11ec-b968-16262ee38c7f           0.0           0.0   
a0dc9659-1c7c-11ec-a91e-16262ee38c7f           0.0           0.0   

Incident                              DRUG_TYPE_17  ...  SYMPTOM_TYPE_6  \
Patient-Uid                                         ...                   
a0dc93f2-1c7c-11ec-9cd2-16262ee38c7f           0.0  ...             0.0   
a0dc94c6-1c7c-11ec-a3a0-16262ee38c7f           0.0  ...             1.0   
a0dc9543-1c7c-11ec-bb63-16262ee38c7f           0.0  ...             1.0   
a0dc95f9-1c7c-11ec-b968-16262ee38c7f           0.0  ...             0.0   
a0dc9659-1c7c-11ec-a91e-16262ee38c7f           0.0  ...             0.0   

Incident                              SYMPTOM_TYPE_7  SYMPTOM_TYPE_8  \
Patient-Uid                                                            
a0dc93f2-1c7c-11ec-9cd2-16262ee38c7f             0.0             0.0   
a0dc94c6-1c7c-11ec-a3a0-16262ee38c7f             0.0             0.0   
a0dc9543-1c7c-11ec-bb63-16262ee38c7f             0.0             0.0   
a0dc95f9-1c7c-11ec-b968-16262ee38c7f             0.0             0.0   
a0dc9659-1c7c-11ec-a91e-16262ee38c7f             0.0             0.0   

Incident                              SYMPTOM_TYPE_9  TARGET DRUG  \
Patient-Uid                                                         
a0dc93f2-1c7c-11ec-9cd2-16262ee38c7f             0.0          0.0   
a0dc94c6-1c7c-11ec-a3a0-16262ee38c7f             0.0          0.0   
a0dc9543-1c7c-11ec-bb63-16262ee38c7f             0.0          0.0   
a0dc95f9-1c7c-11ec-b968-16262ee38c7f             0.0          0.0   
a0dc9659-1c7c-11ec-a91e-16262ee38c7f             0.0          0.0   

Incident                              TEST_TYPE_0  TEST_TYPE_1  TEST_TYPE_2  \
Patient-Uid                                                                   
a0dc93f2-1c7c-11ec-9cd2-16262ee38c7f          0.0          0.0          0.0   
a0dc94c6-1c7c-11ec-a3a0-16262ee38c7f          0.0          0.0          0.0   
a0dc9543-1c7c-11ec-bb63-16262ee38c7f          0.0          0.0          0.0   
a0dc95f9-1c7c-11ec-b968-16262ee38c7f          0.0          0.0          0.0   
a0dc9659-1c7c-11ec-

In [33]:
con_data = con_data.merge(frequency_based, on ="Patient-Uid",how="left")
con_data.head()

Patient-Uid       Date     Incident Date_x Date_y  \
0  a0eb742b-1c7c-11ec-8f61-16262ee38c7f 2020-04-09  TARGET DRUG    NaT    NaT   
1  a0edaf09-1c7c-11ec-a360-16262ee38c7f 2018-06-12  TARGET DRUG    NaT    NaT   
2  a0e9fa0e-1c7c-11ec-8dc7-16262ee38c7f 2019-06-11  TARGET DRUG    NaT    NaT   
3  a0ecc615-1c7c-11ec-aa31-16262ee38c7f 2019-11-15  TARGET DRUG    NaT    NaT   
4  a0ea612f-1c7c-11ec-8cf0-16262ee38c7f 2020-03-18  TARGET DRUG    NaT    NaT   

   DRUG_TYPE_0  DRUG_TYPE_1  DRUG_TYPE_10  DRUG_TYPE_11  DRUG_TYPE_12  ...  \
0          0.0          0.0           0.0           0.0           0.0  ...   
1          0.0          0.0           0.0           0.0           0.0  ...   
2          0.0          0.0           0.0           0.0           0.0  ...   
3          0.0          0.0           0.0           0.0           0.0  ...   
4          0.0          0.0           0.0           0.0           0.0  ...   

   SYMPTOM_TYPE_6  SYMPTOM_TYPE_7  SYMPTOM_TYPE_8  SYMPTOM_TYPE_9  \
0             0.0             0.0             0.0             0.0   
1             0.0             0.0             0.0             0.0   
2             0.0             0.0             0.0             0.0   
3             0.0             0.0             0.0             0.0   
4             0.0             0.0             0.0             0.0   

   TARGET DRUG  TEST_TYPE_0  TEST_TYPE_1  TEST_TYPE_2  TEST_TYPE_3  \
0          1.0          0.0          0.0          0.0          0.0   
1          1.0          0.0          0.0          0.0          0.0   
2          1.0          0.0          0.0          0.0          0.0   
3          1.0          0.0          0.0          0.0          0.0   
4          1.0          0.0          0.0          0.0          0.0   

   TEST_TYPE_4  
0          0.0  
1          0.0  
2          0.0  
3          0.0  
4          0.0  

[5 rows x 53 columns]

In [34]:
con_data["eligibility"]=(pd.to_datetime(con_data["Date"].astype("datetime64[ns]"))- pd.to_datetime(con_data["Date_x"])).dt.days
con_data["eligibility"].value_counts()

Series([], Name: eligibility, dtype: int64)

In [35]:
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report
from sklearn.ensemble import RandomForestClassifier

In [36]:
con_data.columns.value_counts

<bound method IndexOpsMixin.value_counts of Index(['Patient-Uid', 'Date', 'Incident', 'Date_x', 'Date_y', 'DRUG_TYPE_0',
       'DRUG_TYPE_1', 'DRUG_TYPE_10', 'DRUG_TYPE_11', 'DRUG_TYPE_12',
       'DRUG_TYPE_13', 'DRUG_TYPE_14', 'DRUG_TYPE_15', 'DRUG_TYPE_16',
       'DRUG_TYPE_17', 'DRUG_TYPE_2', 'DRUG_TYPE_3', 'DRUG_TYPE_4',
       'DRUG_TYPE_5', 'DRUG_TYPE_6', 'DRUG_TYPE_7', 'DRUG_TYPE_8',
       'DRUG_TYPE_9', 'PRIMARY_DIAGNOSIS', 'SYMPTOM_TYPE_0', 'SYMPTOM_TYPE_1',
       'SYMPTOM_TYPE_10', 'SYMPTOM_TYPE_11', 'SYMPTOM_TYPE_12',
       'SYMPTOM_TYPE_13', 'SYMPTOM_TYPE_14', 'SYMPTOM_TYPE_15',
       'SYMPTOM_TYPE_16', 'SYMPTOM_TYPE_17', 'SYMPTOM_TYPE_18',
       'SYMPTOM_TYPE_19', 'SYMPTOM_TYPE_2', 'SYMPTOM_TYPE_21',
       'SYMPTOM_TYPE_26', 'SYMPTOM_TYPE_29', 'SYMPTOM_TYPE_3',
       'SYMPTOM_TYPE_4', 'SYMPTOM_TYPE_5', 'SYMPTOM_TYPE_6', 'SYMPTOM_TYPE_7',
       'SYMPTOM_TYPE_8', 'SYMPTOM_TYPE_9', 'TARGET DRUG', 'TEST_TYPE_0',
       'TEST_TYPE_1', 'TEST_TYPE_2', 'TEST_TYPE_3', 'T

In [37]:
features = con_data.columns
target ="Incident"

In [38]:
train_data,test_data,train_target,test_target=train_test_split(con_data[features],con_data[target],test_size=0.2,random_state=42)

In [39]:
rf= RandomForestClassifier(random_state=42);
rf.fit(train_data,train_target)

ValueError: could not convert string to float: 'a0ea414d-1c7c-11ec-a6d6-16262ee38c7f'

In [ ]:
#Since the String cannot be used for training the dataset, Label encoder has to be used.

In [51]:
from sklearn.preprocessing import LabelEncoder

label_encoder = LabelEncoder()
train_test_encoded = label_encoder.fit_transform(train_data)
train_target_encoded = label_encoder.fit_transform(train_target)
print(train_target_encoded)
print(train_test_encoded)

ValueError: y should be a 1d array, got an array of shape (14998, 54) instead.

In [49]:
rf= RandomForestClassifier(random_state=42)
rf.fit(train_data,train_target_encoded)

ValueError: could not convert string to float: 'a0ea414d-1c7c-11ec-a6d6-16262ee38c7f'